content/js1/sprints/1/prep/index.md

```diff
@@ -8,4 +8,443 @@ backlog= 'Module-JS1'
 backlog_filter= 'Week 1'
 +++
 
+## Prerequisites
 
+- Basic arithmetic
+- Access to a Linux/MacOS terminal
+
+## Learning objectives
+
+- Define a terminal
+- Describe the purpose of VSCode
+- Define the Node REPL
+- Define an expression
+- Identify the syntactic features of a variable declaration
+- Explain why we use variable declarations
+- Describe what template literals do
+- Create strings using variables and template literals
+- Explain how we use basic arithmetical operators
+- Evaluate basic JavaScript expressions using the Node REPL
+- Predict the result of evaluating an expression in JavaScript
+- Define a function
+- Define a function call
+- Define the term “implementation opaque”
+- Identify the key features of a function call expression: identifier; parentheses, argument
+- Run and print the output of some basic JavaScript functions
+- Execute a JavaScript file using Node
+- Log JavaScript expressions to the terminal
+- Predict the console output when a basic JS program is executed using Node
+
+## Key vocabulary
+
+terminal, console, function, call, invoke, argument, parentheses, REPL, evaluate, implementation opaque
+
+## Notes
+
+### 💻 Interacting with computers
+
+Modern computers are highly complex devices: itt would be too difficult and time-consuming to list all the components that make up a modern computer. To begin understanding, we can start with a simple definition of a computer:
+
+{{<note title="Definition" type="info">}}
+
+A computer is a device used to store and process data. Computers have the capacity to store data through their memory and the ability to change this data through their processing ability.
+
+{{</note>}}
+
+### 🕹️ Using an interface
+
+As a user, we want to interact with computers _without_ understanding how they are built. Every day we interact with machines to carry out specific tasks: in most of these cases we have no idea how these machines are designed and constructed. It would be impossible to make use of new technology if we had to know how all these machines were built at the same time.
+
+Instead we make use of **interfaces**.
+
+{{<note title="Definition" type="info">}}
+
+An **interface** can be thought of as a boundary that allows communication between a user and a machine.
+The user asks the machine to carry out certain tasks via the interface. We can think of the interface as a gateway between the user and the machine.
+
+{{</note>}}
+
+For example, we may want to withdraw money from an ATM machine. We go to a machine in the wall with a screen and a keypad. The **screen and the keypad** form the user interface in this case. We can use this user interface to ask the machine to carry out certain tasks - like give my balance, or please withdraw some money from my account.
+
+{{<note title="Activity" type="info">}}
+
+Try and define the **user interface** for the following devices:
+
+- a calculator
+- a microwave
+- a desktop lamp
+- Alexa
+- ChatGPT
```

Comment:
Might be interesting to introduce a website here? (I suppose ChatGPT is, but a different kettle of fish).

I was thinking that Facebook might be an interesting example?